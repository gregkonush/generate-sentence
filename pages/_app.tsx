import type { AppProps } from 'next/app'
import GlobalStyles from '../components/global-styles'

const App = ({ Component, pageProps }: AppProps): JSX.Element => (
  <div>
    <GlobalStyles />
    <Component {...pageProps} />
  </div>
)

export default App
