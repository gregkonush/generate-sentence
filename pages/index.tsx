import * as React from 'react'
import useSWR from 'swr'
import Head from 'next/head'
import 'twin.macro'
import { useForm } from 'react-hook-form'
import { Input, Label } from '@components'

const fetcher = (url) => fetch(url).then((r) => r.json())

export const Home = (): JSX.Element => {
  const { register, watch } = useForm({
    defaultValues: { subject: 'dog', verb: 'eat', object: 'water', tense: 'present', progressive: false },
  })
  const { data } = useSWR(
    [
      `https://lt-nlgservice.herokuapp.com/rest/english/realise?subject=${watch('subject')}&verb=${watch(
        'verb'
      )}&object=${watch('object')}&tense=${watch('tense')}&${watch('progressive') ? 'progressive=progressive' : ''}`,
      watch('subject'),
      watch('verb'),
      watch('object'),
    ],
    fetcher,
    { dedupingInterval: 100 }
  )

  return (
    <div tw="h-screen bg-gray-900 text-gray-200 p-10">
      <Head>
        <title>Complete your sentence</title>
      </Head>
      <form tw="flex flex-col space-y-3 md:w-2/3 lg:w-1/2 2xl:w-1/3">
        <Input register={register} name="subject" label="Enter a subject" />
        <Input register={register} name="verb" label="Enter a verb" />
        <Input register={register} name="object" label="Enter an object" />
        <Label htmlFor="tense" text="Choose a tense" />
        <select tw="p-2 text-gray-700 space-y-0" {...register('tense')}>
          <option>past</option>
          <option>present</option>
          <option>future</option>
        </select>
        <div tw="flex space-x-2 items-center">
          <input type="checkbox" {...register('progressive')} />
          <Label htmlFor="progressive" text="progressive?" />
        </div>
        <hr tw="border-gray-700" />
      </form>
      <div tw="mt-3">Result:</div>
      {data?.sentence ? (
        <div tw="text-blue-300 text-2xl">{data.sentence}</div>
      ) : (
        <svg tw="animate-spin h-5 w-5 m-2" viewBox="0 0 24 24">
          <circle tw="opacity-25" cx="12" cy="12" r="10" stroke="currentColor" strokeWidth="4" />
          <path
            tw="opacity-75"
            fill="currentColor"
            d="M4 12a8 8 0 018-8V0C5.373 0 0 5.373 0 12h4zm2 5.291A7.962 7.962 0 014 12H0c0 3.042 1.135 5.824 3 7.938l3-2.647z"
          />
        </svg>
      )}
    </div>
  )
}

export default Home
