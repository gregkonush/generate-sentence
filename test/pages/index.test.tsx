import React from 'react'
import { render } from '../testUtils'
import { Home } from '../../pages/index'

describe('Home page', () => {
  it('matches snapshot', () => {
    const { container } = render(<Home />, {})
    expect(container).toMatchSnapshot()
  })
})
