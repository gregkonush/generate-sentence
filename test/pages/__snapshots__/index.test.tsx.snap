// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page matches snapshot 1`] = `
<div>
  <div
    class="pages___StyledDiv-sc-1rar0h2-0 PjuDW"
  >
    <form
      class="pages___StyledForm-sc-1rar0h2-1 kiDKwk"
    >
      <div
        class="input___StyledDiv-sc-6wa7t4-0 eMbqot"
      >
        <label
          class="label___StyledLabel-sc-1sm73mb-0 bMVZEK"
          for="subject"
        >
          Enter a subject
        </label>
        <input
          class="input___StyledInput-sc-6wa7t4-1 enXCtp"
          name="subject"
        />
      </div>
      <div
        class="input___StyledDiv-sc-6wa7t4-0 eMbqot"
      >
        <label
          class="label___StyledLabel-sc-1sm73mb-0 bMVZEK"
          for="verb"
        >
          Enter a verb
        </label>
        <input
          class="input___StyledInput-sc-6wa7t4-1 enXCtp"
          name="verb"
        />
      </div>
      <div
        class="input___StyledDiv-sc-6wa7t4-0 eMbqot"
      >
        <label
          class="label___StyledLabel-sc-1sm73mb-0 bMVZEK"
          for="object"
        >
          Enter an object
        </label>
        <input
          class="input___StyledInput-sc-6wa7t4-1 enXCtp"
          name="object"
        />
      </div>
      <label
        class="label___StyledLabel-sc-1sm73mb-0 bMVZEK"
        for="tense"
      >
        Choose a tense
      </label>
      <select
        class="pages___StyledSelect-sc-1rar0h2-2 hvnios"
        name="tense"
      >
        <option>
          past
        </option>
        <option>
          present
        </option>
        <option>
          future
        </option>
      </select>
      <div
        class="pages___StyledDiv2-sc-1rar0h2-3 fFkQxp"
      >
        <input
          name="progressive"
          type="checkbox"
        />
        <label
          class="label___StyledLabel-sc-1sm73mb-0 bMVZEK"
          for="progressive"
        >
          progressive?
        </label>
      </div>
      <hr
        class="pages___StyledHr-sc-1rar0h2-4 Tiivl"
      />
    </form>
    <div
      class="pages___StyledDiv3-sc-1rar0h2-5 jgJrUc"
    >
      Result:
    </div>
    <svg
      class="pages___StyledSvg-sc-1rar0h2-7 kIxLwG"
      viewBox="0 0 24 24"
    >
      <circle
        class="pages___StyledCircle-sc-1rar0h2-8 jWHHNh"
        cx="12"
        cy="12"
        r="10"
        stroke="currentColor"
        stroke-width="4"
      />
      <path
        class="pages___StyledPath-sc-1rar0h2-9 jdnYLu"
        d="M4 12a8 8 0 018-8V0C5.373 0 0 5.373 0 12h4zm2 5.291A7.962 7.962 0 014 12H0c0 3.042 1.135 5.824 3 7.938l3-2.647z"
        fill="currentColor"
      />
    </svg>
  </div>
</div>
`;
